tests: 'make distclean' error

## How to reproduce
Use latest master 01a5aadeabb76d9eb397f755a9d79f7d79181a8f.

```
cd /tmp
git clone --recursive https://github.com/sheepdog/sheepdog
cd sheepdog

./autogen.sh
./configure --disable-corosync --enable-unittest && make check && make clean

make distclean
```

## Expected behavior
'make distclean' succeeded.

## Actual behavior
'make distclean' failed in tests/unit/sheep.

```
[...snipped...]
Making distclean in tests/unit
make[1]: Entering directory '/tmp/sheepdog/tests/unit'
Making distclean in mock
make[2]: Entering directory '/tmp/sheepdog/tests/unit/mock'
[...snipped...]
make[2]: Leaving directory '/tmp/sheepdog/tests/unit/mock'
Making distclean in dog
make[2]: Entering directory '/tmp/sheepdog/tests/unit/dog'
[...snipped...]
make[2]: Leaving directory '/tmp/sheepdog/tests/unit/dog'
Making distclean in sheep
make[2]: Entering directory '/tmp/sheepdog/tests/unit/sheep'
Makefile:786: ../../../sheep/.deps/config.Po: No such file or directory
Makefile:787: ../../../sheep/.deps/gateway.Po: No such file or directory
Makefile:788: ../../../sheep/.deps/group.Po: No such file or directory
Makefile:789: ../../../sheep/.deps/migrate.Po: No such file or directory
Makefile:790: ../../../sheep/.deps/object_list_cache.Po: No such file or directory
Makefile:791: ../../../sheep/.deps/ops.Po: No such file or directory
Makefile:792: ../../../sheep/.deps/recovery.Po: No such file or directory
Makefile:793: ../../../sheep/.deps/request.Po: No such file or directory
Makefile:794: ../../../sheep/.deps/vdi.Po: No such file or directory
Makefile:795: ../../../sheep/store/.deps/common.Po: No such file or directory
Makefile:796: ../../../sheep/store/.deps/md.Po: No such file or directory
make[2]: *** No rule to make target '../../../sheep/store/.deps/md.Po'.  Stop.
make[2]: Leaving directory '/tmp/sheepdog/tests/unit/sheep'
Makefile:384: recipe for target 'distclean-recursive' failed
make[1]: *** [distclean-recursive] Error 1
make[1]: Leaving directory '/tmp/sheepdog/tests/unit'
Makefile:496: recipe for target 'distclean-recursive' failed
make: *** [distclean-recursive] Error 1
```

--- a/tests/unit/lib/Makefile.am
+++ b/tests/unit/lib/Makefile.am
@@ -17,26 +17,39 @@ LIBS			= -pthread -lm					\
 			  $(top_srcdir)/lib/libsd.a			\
 			  @CHECK_LIBS@
 
-test_util_SOURCES	= test_util.c ../../../lib/util.c		\
+test_util_SOURCES	= test_util.c lib/util.c		\
 			  ../mocks/Mocklogger.c				\
-			  ../cmock/src/cmock.c ../unity/src/unity.c
+			  cmock.c unity.c
 
-test_work_SOURCES	= test_work.c ../../../lib/work.c		\
-			  ../unity/src/unity.c
+test_work_SOURCES	= test_work.c lib/work.c		\
+			  unity.c
 
 test_punchhole_SOURCES	= test_punchhole.c				\
-			  ../../../lib/util.c				\
+			  lib/util.c				\
 			  ../mocks/Mocklogger.c				\
-			  ../cmock/src/cmock.c ../unity/src/unity.c
+			  cmock.c unity.c
 
 test_atomic_create_and_write_SOURCES =					\
 			  test_atomic_create_and_write.c		\
-			  ../../../lib/common.c				\
+			  lib/common.c				\
 			  ../mocks/Mocklogger.c				\
-			  ../cmock/src/cmock.c ../unity/src/unity.c
+			  cmock.c unity.c
 
 clean-local:
 	rm -f ${check_PROGRAMS} *.o lib.info
 
 coverage:
 	@lcov -d . -c -o lib.info
+
+lib/%.c: $(top_srcdir)/lib/%.c
+	@mkdir -p $(@D)
+	@cp $< $@
+
+cmock.c: ../cmock/src/cmock.c
+	@cp $< $@
+
+unity.c: ../unity/src/unity.c
+	@cp $< $@
+
+distclean-local:
+	rm -rf lib cmock.c unity.c

--- a/tests/unit/sheep/Makefile.am
+++ b/tests/unit/sheep/Makefile.am
@@ -19,7 +19,7 @@ LIBS			= $(top_srcdir)/lib/libsd.a		\
 			  @CHECK_LIBS@
 
 test_vdi_SOURCES	= test_vdi.c sheep/vdi.c mock_sheep.c mock_store.c		\
-			  mock_request.c ../unity/src/unity.c
+			  mock_request.c unity.c
 
 test_cluster_driver_SOURCES	= mock_sheep.c mock_group.c		\
 				  sheep/cluster/local.c	\
@@ -35,33 +35,33 @@ endif
 test_hash_SOURCES	= test_hash.c mock_sheep.c mock_group.c \
 				mock_plain_store.c mock_gateway.c mock_store.c mock_vdi.c
 
-test_group_SOURCES	= test_group.c ../../../sheep/group.c \
-				../../../sheep/ops.c \
+test_group_SOURCES	= test_group.c sheep/group.c \
+				sheep/ops.c \
 				mock_sheep.c \
-				../../../sheep/request.c \
-				../../../sheep/store/common.c \
-				../../../sheep/store/md.c \
-				../../../sheep/vdi.c \
-				../../../sheep/config.c \
-				../../../sheep/recovery.c \
-				../../../sheep/gateway.c \
-				../../../sheep/object_list_cache.c \
-				../../../sheep/migrate.c \
-				../cmock/src/cmock.c ../unity/src/unity.c
+				sheep/request.c \
+				sheep/store/common.c \
+				sheep/store/md.c \
+				sheep/vdi.c \
+				sheep/config.c \
+				sheep/recovery.c \
+				sheep/gateway.c \
+				sheep/object_list_cache.c \
+				sheep/migrate.c \
+				cmock.c unity.c
 
-test_recovery_SOURCES   = test_recovery.c ../../../sheep/recovery.c \
-                ../../../sheep/ops.c \
+test_recovery_SOURCES   = test_recovery.c sheep/recovery.c \
+                sheep/ops.c \
                 mock_sheep.c \
-                ../../../sheep/request.c \
-                ../../../sheep/store/common.c \
-                ../../../sheep/store/md.c \
-                ../../../sheep/vdi.c \
-                ../../../sheep/config.c \
-                ../../../sheep/group.c \
-                ../../../sheep/gateway.c \
-                ../../../sheep/object_list_cache.c \
-                ../../../sheep/migrate.c \
-                ../cmock/src/cmock.c ../unity/src/unity.c
+                sheep/request.c \
+                sheep/store/common.c \
+                sheep/store/md.c \
+                sheep/vdi.c \
+                sheep/config.c \
+                sheep/group.c \
+                sheep/gateway.c \
+                sheep/object_list_cache.c \
+                sheep/migrate.c \
+                cmock.c unity.c
 
 clean-local:
 	rm -f ${check_PROGRAMS} *.o sheep.info
@@ -73,5 +73,11 @@ sheep/%.c: $(top_srcdir)/sheep/%.c
 	@mkdir -p $(@D)
 	@cp $< $@
 
+cmock.c: ../cmock/src/cmock.c
+	@cp $< $@
+
+unity.c: ../unity/src/unity.c
+	@cp $< $@
+
 distclean-local:
-	rm -rf sheep
+	rm -rf sheep cmock.c unity.c

